packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -38,6 +38,13 @@ export interface BucketDeploymentProps {
    */
   readonly destinationKeyPrefix?: string;
 
+  /**
+   * If this is set, the zip file will be synced to the destination S3 bucket and unzipped.
+   * If false, the file will remain zipped in the destination bucket.
+   * @default true
+   */
+  readonly unzipFile?: boolean;
```

Comment:
#8065 discusses alternatives for deploying a zipped asset. Specifically  https://github.com/aws/aws-cdk/issues/8065#issuecomment-632610526

Have you considered alternatives and why have you chosen this approach?

Reply:
Yes, a separate PR will be drafted for the s3-deployment changes that clarifies design considerations and alternatives. The feature addressed by this PR relies on these s3 changes and given that we are working against a deadline, we submitted this PR as a draft to get early feedback solely on the changes made to **aws-servicecatalog**. Hope this helps to clarify.

Reply:
@mrgrain Just to clarify, are you asking for design considerations and alternatives for the entire PR or just for the S3 changes?

Reply:
I would say the entire PR. This is a pretty big deviation from how we handle assets in the CDK. It looks like you are essentially creating a shadow assets mechanism that we will have to support going forward.